src/BaseModel.php

```diff
@@ -590,31 +582,36 @@ public function lastError(): Error
     }
 
     /**
-     * Invoke callback
+     * Invoke hook
      *
-     * Invokes the the callback in the order that they are stored in the callback
-     * array for a passed in callback type. All additional parameters are sent to
-     * the invoked callback.
+     * Invokes the hook specified by the name. The whole hook name consists of string model,
+     * class name or custom hook name stored in the private property hookName, before or after 
+     * concatenated with the modelMethod. Example: "model.user.before.init" for a user model.
      *
-     * @param string $name Name of the callback
-     * @param bool $before Invoke 'before' callables of '$name' callback, default self::CALLBACK_BEFORE
+     * @param string $modelMethod Model method name.
+     * @param bool $before Invoke 'before' '$modelMethod' hook, default self::HOOK_BEFORE
      * @return void
      */
-    protected function invokeCallback(string $name, bool $before = self::CALLBACK_BEFORE)
+    protected function invokeHook(string $modelMethod, bool $before = self::HOOK_BEFORE)
     {
-        $name = ($before ? "before" : "after") . ucfirst($name);
-        if (isset($this->{$name}) === false || is_array($this->{$name}) === false) {
-            // @todo: throw exception
-            return;
-        }
-
-        $params = array_slice(func_get_args(), 2);
-        foreach ($this->{$name} as $callable) {
-            $callable(...$params);
+        $clsName = $this->hookName;
+        if ($clsName === null) {
+            $clsName = $this->getClassName();
         }
+        $name = "model" . $clsName . ($before ? "before" : "after") . $modelMethod;
```

Comment:
Add missing commas(.) to the name. The end result must be **model.classname.before.modelMethod**